drivers/flash/soc_flash_cc13xx_cc26xx.c

```diff
@@ -183,17 +184,29 @@ static int flash_cc13xx_cc26xx_write(const struct device *dev, off_t offs,
 	 *
 	 * The pui8DataBuffer pointer can not point to flash.
 	 */
+#if K_HEAP_MEM_POOL_SIZE
```

Comment:
Hmm... It's fairly uncommon to dynamically allocate in drivers in Zephyr.

It might make sense to create an array of some fixed size in the `struct data` driver object, since that will always live in ram, and then initialize that location in the driver `init()` with contents that would be in devicetree / rom.